### PR TITLE
Windows build + CI

### PR DESCRIPTION
**Pull request by vitalyster**

### CODE REVIEW

File: src/automaton/core/interop/ethereum/eth_helper_functions.h [hunk 1]
```diff
@@ -34,7 +36,7 @@ static size_t curl_callback(void *contents, size_t size, size_t nmemb, std::stri
     s->append(reinterpret_cast<char*>(contents), new_length);
   }
   catch (std::bad_alloc &e) {
-    LOG(ERROR) << "Bad_alloc while reading data!";
+    LOG(WARNING) << "Bad_alloc while reading data!";
```
Comment (akovachev) [inline src/automaton/core/interop/ethereum/eth_helper_functions.h]: We should revert all the ERROR -> WARNING changes and do a #define if necessary.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -184,12 +195,14 @@ macro(automaton_test test_path test_name)
...
     automaton/tests/data/proto_files.cc
   )
+  target_include_directories(${test_name} PRIVATE ${GTEST_INCLUDE_DIR})
   target_link_libraries(${test_name} automaton-core ${GTEST_LIBS})
   add_test(
     NAME ${test_name}
     COMMAND ${test_name}
     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
   )
+  set_property(TARGET ${test_name} PROPERTY VS_DEBUGGER_WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}")
```
Comment (akovachev) [inline src/CMakeLists.txt]: Need to add this for some of the other targets like the automaton-core-cli.
(And the above WORKING_DIRECTORY parameter)
Might be best to create a custom target macro that creates an easy to debug target which also links statically all the libs. (as I understand that's already the case though - the static linking)

**Comment by akovachev** [comment]
Also we can remove Travis/AppVeyor (or at least disable)

File: src/CMakeLists.txt (hunk 2 above)
Comment (vitalyster) [reply src/CMakeLists.txt]: Macro added

**Comment by vitalyster** [comment]
Appveyor and Travis hooks should be disabled in project settings

**Pull request merged**